### PR TITLE
Access control: service account role check

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
**What this PR does / why we need it**:
* forbids users from creating service accounts and API keys with higher permissions than what the user has;
* allows non-admin users to target API key endpoints if they have the right permissions.
* remove the roles attribute in listing of a specific SAT.
* assign the Viewer role upon creation of a service account.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/3161
Fixes https://github.com/grafana/grafana-enterprise/issues/3160

### CODE REVIEW

**Review by vtorosyan — APPROVED** [review]

File: pkg/api/api.go [hunk 1]
```diff
@@ -279,7 +279,7 @@ func (hs *HTTPServer) registerRoutes() {
 			keysRoute.Get("/", authorize(reqOrgAdmin, ac.EvalPermission(ac.ActionAPIKeyRead, ac.ScopeAPIKeysAll)), routing.Wrap(hs.GetAPIKeys))
 			keysRoute.Post("/", authorize(reqOrgAdmin, ac.EvalPermission(ac.ActionAPIKeyCreate)), quota("api_key"), routing.Wrap(hs.AddAPIKey))
 			keysRoute.Delete("/:id", authorize(reqOrgAdmin, ac.EvalPermission(ac.ActionAPIKeyDelete, apikeyIDScope)), routing.Wrap(hs.DeleteAPIKey))
-		}, reqOrgAdmin)
```
Comment (eleijonmarck) [inline pkg/api/api.go:282]: this is just clean up i guess

Comment (IevaVasiljeva) [reply pkg/api/api.go:282]: I think this is a bug from when RBAC was added to API keys. It means that legacy access control was still applied to all API key endpoints, so a non-admin user with the correct RBAC permissions could not access them.

File: pkg/api/apikey.go [hunk 2]
```diff
@@ -70,6 +70,9 @@ func (hs *HTTPServer) AddAPIKey(c *models.ReqContext) response.Response {
 	if !cmd.Role.IsValid() {
 		return response.Error(400, "Invalid role specified", nil)
 	}
+	if !c.OrgRole.Includes(cmd.Role) {
+		return response.Error(http.StatusForbidden, "Cannot assign a role higher than user's role", nil)
+	}
```
Comment (eleijonmarck) [inline pkg/api/apikey.go:75]: this was also there for API keys as well? 🤔

**Review by eleijonmarck — APPROVED** [review]
LGTM

**Pull request merged**